example/spec/helpers/granuleUtils.js

```diff
@@ -144,11 +146,120 @@ async function setupTestGranuleForIngest(bucket, inputPayloadJson, granuleRegex,
+ * @param {string} inputPayloadJson - input payload as a JSON string
+ * @param {number} granuleCountPerPayload - number of granules per workflow
+ * @param {string} granuleRegex - regex to generate the new granule id
+ * @param {string} testSuffix - suffix for test-specific collection
+ * @param {string} testDataFolder - test data S3 path. Only needed when isPassThroughWorkflow = false
+ * @returns {Promise<Object>} - input payload as a JS object with the updated granule ids
+ */
+async function setupGranulesForIngestLoadTest(isPassThroughWorkflow = true, bucket, inputPayloadJson, granuleCountPerPayload, granuleRegex, testSuffix = '', testDataFolder = undefined) {
+  let payloadJson;
+
+  if (testDataFolder) {
+    payloadJson = replace(
+      new RegExp('cumulus-test-data/pdrs', 'g'),
+      testDataFolder,
+      inputPayloadJson
+    );
+  } else {
+    payloadJson = inputPayloadJson;
+  }
+  const baseInputPayload = JSON.parse(payloadJson);
+  baseInputPayload.granules[0].dataType += testSuffix;
+  const granuleToReplicate = baseInputPayload.granules.shift();
+
+  console.log(blueConsoleLog(), `\n===== Create ${granuleCountPerPayload} granules per payload =====`);
+  const createS3GranuleFilesPromises = new Array(granuleCountPerPayload).fill().map(
+    async () => {
+      const replicatedGranule = await generatePayloadGranule(isPassThroughWorkflow, granuleToReplicate, bucket, granuleRegex);
+      baseInputPayload.granules.push(replicatedGranule);
+    }
+  );
+  await Promise.all(createS3GranuleFilesPromises);
+
+  if (isPassThroughWorkflow === false) {
+    console.log(blueConsoleLog(), `===== Uploaded ${granuleCountPerPayload * granuleToReplicate.files.length} granule files per payload =====\n`);
+  }
+
+  return baseInputPayload;
+}
+
+const cleanupLoadTestGranules = async (stackName, granuleList) => {
+  const granIds = granuleList.map((g) => g.granuleId);
+  const bulkDeleteResponse = await granules.bulkDeleteGranules({ prefix: stackName, body: { ids: granIds } });
+  const verifiedResponse = apiTestUtils.verifyCumulusApiResponse(bulkDeleteResponse, [202]);
+  const responseBody = JSON.parse(verifiedResponse.body);
+  console.log(blueConsoleLog(), `\n===== Bulk Granule deleting ${granIds.length}. Async operation ID: ${JSON.stringify(responseBody)} =====`);
+
+  await pWaitFor(async () => {
+    let asyncOperation = {};
+    try {
+      asyncOperation = await apiTestUtils.getAsyncOperation({
+        prefix: stackName,
+        id: responseBody.id,
+      });
+    } catch (error) {
+      return false;
+    }
+    if (!asyncOperation.body) {
+      return false;
+    }
+    if (['RUNNER_FAILED', 'TASK_FAILED'].includes(JSON.parse(asyncOperation.body).status)) {
+      throw new Error(`Cleanup failed on delete granules async operation ${JSON.stringify(asyncOperation.body)}`);
+    }
+    return JSON.parse(asyncOperation.body).status === 'SUCCEEDED';
+  }, { interval: 60 * 1000, timeout: 15 * 60 * 1000 });
```

Comment:
What's up with the `{ interval: 60 * 1000, timeout: 15 * 60 * 1000 }`?

Reply:
These are the Options params for node_models > pWaitFor, as in Number of milliseconds to wait before retrying and automatically
rejecting.